src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java

```diff
@@ -0,0 +1,262 @@
+            }
+            worldProvider.setBlock(bushComponent.position, blockManager.getBlock(BlockManager.AIR_ID));
+        }
+
+        /* Drop some seeds */
+        for (int i = 0; i < numSeeds; i++) {
+            Prefab seed = bushComponent.seed == null ? bushComponent.produce : bushComponent.seed;
+            EntityRef seedItem = entityManager.create(seed);
+            Vector3f position = bushComponent.position.toVector3f().add(0, 0.5f, 0);
+            seedItem.send(new DropItemEvent(position));
+            seedItem.send(new ImpulseEvent(random.nextVector3f(30.0f)));
+        }
+
+
+    }
+
+    /**
+     * Called when the bush should grow
+     *
+     * @param event         The event indicating the timer has ended
+     * @param bush          The bush to grow
+     * @param bushComponent The bush's definition
+     */
+    @ReceiveEvent
+    public void onBushGrowth(DelayedActionTriggeredEvent event, EntityRef bush, BushDefinitionComponent bushComponent) {
+        EntityRef newBush = null;
+        if (bushComponent.currentStage < bushComponent.stages.length - 1) {
+            newBush = doBushGrowth(bushComponent, 1);
+        }
+        resetDelay(newBush == null ? bush : newBush,
+                bushComponent.stages[bushComponent.currentStage].minTime,
+                bushComponent.stages[bushComponent.currentStage].maxTime);
+    }
+
+    /**
+     * Called when an attempt to harvest the bush is made.
+     * If the bush is
+     *
+     * @param event  The activation event
+     * @param entity The entity doing the harvesting
+     */
+    @ReceiveEvent
+    public void onHarvest(ActivateEvent event, EntityRef entity) {
+        EntityRef target = event.getTarget();
+        EntityRef harvester = entity.equals(target) ? event.getInstigator() : entity;
+        /* Ensure the target is a plant and the entities are valid */
+        if (!event.isConsumed() && target.exists() && harvester.exists()
+                && target.hasComponent(BushDefinitionComponent.class)) {
+            BushDefinitionComponent bushComponent = target.getComponent(BushDefinitionComponent.class);
+
+            /* Produce is only given in the final stage */
+            if (bushComponent.currentStage == bushComponent.stages.length - 1) {
+                EntityRef produce;
+                /* Handle a dodgy produce */
+                try {
+                    produce = entityManager.create(bushComponent.produce);
+                    boolean giveSuccess = inventoryManager.giveItem(harvester, target, produce);
+                    if (!giveSuccess) {
+                        Vector3f position = event.getTargetLocation().add(0, 0.5f, 0);
+                        produce.send(new DropItemEvent(position));
+                        produce.send(new ImpulseEvent(random.nextVector3f(15.0f)));
+                    }
+                } catch (NullPointerException e) {
+                    logger.error("Unable to create produce: " + bushComponent.produce);
```

Comment:
don't catch nullpointerexceptions, do a null check if needed (logging in the else-case is okay though).